src/test/renderWithProviders.tsx

```diff
@@ -387,13 +393,52 @@ export function setupFakeApp({ queries, mutations }: SetupFakeAppProps = {}) {
   };
 }
 
+function injectFormDataSavingSimulator(
+  queryMocks: AppQueries,
+  mutationMocks: AppMutations,
+  mockBackend: Required<ExtendedRenderOptions>['mockFormDataSaving'],
+) {
+  const models: Record<string, unknown> = {};
+  const originalFetchFormData = queryMocks.fetchFormData;
+
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  (queryMocks as any).fetchFormData = jest.fn().mockImplementation(async (url: string) => {
+    const result = await originalFetchFormData(url);
+    models[url] = result;
+    return result;
+  });
+
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  (mutationMocks as any).doPatchFormData = jest
+    .fn()
+    .mockImplementation(async (url: string, req: IDataModelPatchRequest): Promise<IDataModelPatchResponse> => {
+      const model = structuredClone(models[url] ?? {});
+      applyPatch(model, req.patch);
+      const afterProcessing = typeof mockBackend === 'function' ? mockBackend(model, url) : model;
+      models[url] = afterProcessing;
+
+      return {
+        newDataModel: afterProcessing as object,
+        validationIssues: {},
+      };
+    });
+
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  (mutationMocks as any).doPostStatelessFormData = jest.fn().mockImplementation(async (url: string, data: unknown) => {
+    const afterProcessing = typeof mockBackend === 'function' ? mockBackend(data, url) : data;
+    models[url] = afterProcessing;
+    return afterProcessing;
+  });
+}
+
```

Comment:
This looks very useful! Is there a reason multi-patch does not have a mock implementation? I guess we dont have any unit tests with multiple data models yet

Reply:
Yep, that's about it. Shouldn't be any issues adding support for it, but as it stands I didn't bother looking into the nitty gritty.. :face_with_peeking_eye: